magellan-library/src/main/java/com/wealthfront/magellan/navigation/NavigationDelegate.kt

```diff
@@ -160,4 +182,39 @@ class NavigationDelegate(
   }
 
   private fun atRoot() = backStack.size <= 1
+
+  private fun setupCurrentScreenToBeShown(currentNavigable: NavigableCompat) {
+    if (currentNavigable is Journey<*>) {
+      menu?.let { currentNavigable.setMenu(it) }
+    }
+    currentNavigable.setTitle(currentNavigable.getTitle(activity!!))
+    updateMenu(menu, currentNavigable)
+    callOnNavigate(currentNavigable)
+  }
+
+  private fun updateMenu(menu: Menu?, navItem: NavigableCompat? = null) {
+    // Need to post to avoid animation bug on disappearing menu
+    Handler(Looper.getMainLooper()).post {
+      menu?.let {
+        for (i in 0 until menu.size()) {
+          menu.getItem(i).isVisible = false
+          (rootNavigable as? ActionBarModifier)?.onUpdateMenu(menu)
+          rootNavigable.childNavigables().filterIsInstance(ActionBarModifier::class.java).forEach { it.onUpdateMenu(menu) }
+          (navItem as? ActionBarModifier)?.onUpdateMenu(menu)
+          navItem?.childNavigables()?.filterIsInstance(ActionBarModifier::class.java)?.forEach { it.onUpdateMenu(menu) }
```

Comment:
Wait, we want to do this for every menu item? I think we want to move this out of the for loop

Reply:
Moved out of the loop.